yt/frontends/tipsy/io.py

```diff
@@ -423,22 +425,37 @@ def _identify_fields(self, data_file):
         return self._field_list, {}
 
     def _calculate_particle_offsets(self, data_file, pcounts):
+        # This computes the offsets for each particle type into a "data_file."  Note that 
+        # the term "data_file" here is a bit overloaded, and also refers to a
+        # "chunk" of particles inside a data file.
+        # data_file.start represents the *particle count* that we should start at.
+        #
+        # At this point, data_file will have the total number of particles
+        # that this chunk represents located in the property total_particles.
+        # Because in tipsy files the particles are stored sequentially, we can
+        # figure out where each one starts.
+        # We first figure out the global offsets, then offset them by the count
+        # and size of each individual particle type.
         field_offsets = {}
+        # Initialize pos to the point the first particle type would start
         pos = data_file.ds._header_offset
+        global_offsets = {}
+        field_offsets = {}
         for i, ptype in enumerate(self._ptypes):
             if ptype not in self._pdtypes:
+                # This means we don't have any, I think, and so we shouldn't
+                # stick it in the offsets.
                 continue
+            # Note that much of this will be computed redundantly; future
+            # refactorings could fix this.
+            global_offsets[ptype] = pos
             size = self._pdtypes[ptype].itemsize
-            field_offsets[ptype] = pos
             params = self.ds.parameters
```

Comment:
This line is triggering the `flake8` error since `params` is no longer used.

Reply:
Should be fixed now!